source/common/statistic_impl.cc

```diff
@@ -148,13 +171,21 @@ void HdrStatistic::addValue(uint64_t value) {
   // or maximum value we passed when initializing histogram_.
   if (!hdr_record_value(histogram_, value)) {
     ENVOY_LOG(warn, "Failed to record value into HdrHistogram.");
+  } else {
+    StatisticImpl::addValue(value);
   }
 }
 
+// We override count for the Hdr statistics, because it may have dropped
+// out of range values. hence our own tracking may be inaccurate.
 uint64_t HdrStatistic::count() const { return histogram_->total_count; }
 double HdrStatistic::mean() const { return count() == 0 ? std::nan("") : hdr_mean(histogram_); }
 double HdrStatistic::pvariance() const { return pstdev() * pstdev(); }
 double HdrStatistic::pstdev() const { return count() == 0 ? std::nan("") : hdr_stddev(histogram_); }
+uint64_t HdrStatistic::min() const {
+  return count() == 0 ? UINT64_MAX : hdr_value_at_percentile(histogram_, 0);
+}
+uint64_t HdrStatistic::max() const { return hdr_value_at_percentile(histogram_, 100); }
```

Comment:
Will max() blow up if count()==0?

Reply:
We should be ok with respect to that (see `TypedStatisticTest::Empty`)